aws/resource_aws_flow_log.go

```diff
@@ -136,17 +142,16 @@ func resourceAwsLogFlowRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("traffic_type", fl.TrafficType)
 	d.Set("log_group_name", fl.LogGroupName)
 	d.Set("iam_role_arn", fl.DeliverLogsPermissionArn)
+	d.Set("log_destination", fl.LogDestination)
+	d.Set("log_destination_type", fl.LogDestinationType)
+	d.Set("resource_id", fl.ResourceId)
 
-	var resourceKey string
 	if strings.HasPrefix(*fl.ResourceId, "vpc-") {
```

Comment:
I have been thinking of various places this kind of code is used/useful (e.g. I've been thinking about solutions for https://github.com/terraform-providers/terraform-provider-aws/issues/26).
Do you think we should add some generic EC2 resource ID parsing function that returns the resource type from the ID (e.g. based on rules [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/resource-ids.html))?